### PR TITLE
Running slowly is the same as walking.

### PR DESCRIPTION
**Pull request by nullbear**
:cl:
:tweak: Certain objects and effects that were psychically able to tell whether
you have the intent to walk or run, are now triggered instead by whether
you're actually walking or running.
/:cl:

This means that as long as you are moving at or slower than walking
speed that you will not slip on wet floors or trigger portable flashers.

Limping slowly in run-intent no longer triggers run-intent stuff.
Moving really fast while in walk-intent should **NOT** trigger run-intent stuff.

### CODE REVIEW

File: code/__DEFINES/mobs.dm [hunk 1]
```diff
@@ -11,6 +11,9 @@
 #define MOVE_INTENT_WALK "walk"
 #define MOVE_INTENT_RUN  "run"
 
+//Walking Speed Threshold for slips and stuff. If you move slower than this, you are considered walking, regardless of intent.
+#define MOVE_WALK_THRESHOLD 4 // Roundstart with shoes in walk-intent your move-delay reads as 4.
+
```
Comment (MrPerson) [inline code/__DEFINES/mobs.dm]: Wouldn't it make more sense to have this == the actual slowdown applied by walk intent?

Comment (nullbear) [reply code/__DEFINES/mobs.dm]: Yea i wanted to ask:

Should i do: 
#define MOVE_WALK_THRESHOLD CONFIG_GET(WALK_DELAY)

or should i get rid of MOVE_WALK_THRESHOLD entirely and just use CONFIG_GET(WALK_DELAY)

**Comment by ATH1909** [comment]
*flashbacks of getting flashed by autoflashers despite being on walk intent*

I really, really don't want my terrible experiences with autoflashers to start coming from everything that requires walk intent to safely bypass, hence my downthumb.

**Comment by MrPerson** [comment]
Have walk intent force you to move at the "walking speed threshhold" or your current speed, whichever is faster. That solves all problems 100%.

**Comment by ATH1909** [comment]
> Have walk intent force you to move at the "walking speed threshhold" or your current speed, whichever is faster. That solves all problems 100%.

Whichever is SLOWER, not whichever is FASTER, right? Otherwise, you get people who can move faster while walking than while running.

**Comment by nullbear** [comment]
Changed it so that walk-intent limits your move speed to the safe walking speed.

**Comment by MrPerson** [comment]
Yeah sorry, it was really late ha ha

File: code/__DEFINES/mobs.dm (hunk 1 above)
Comment (Cyberboss) [reply code/__DEFINES/mobs.dm]: latter

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Remove the define in favor of CONFIG_GET

**Comment by anconfuzedrock** [comment]
guy moving as fast as he physically can in a hardsuit should not be able to avoid slipping, dumb. people with ONE FOOT immune to slipping, dumb

File: code/__DEFINES/mobs.dm (hunk 1 above)
Comment (nullbear) [reply code/__DEFINES/mobs.dm]: Done

**Comment by Bawhoppen** [comment]
Confuzed raises a good point. Running intent represents the sacrifice of stability in favor of speed. If you are severely crippled but still trying to go as quickly as you can, you are still taking an increased risk of losing your balance.

**Comment by nullbear** [comment]
@Bawhoppen I was planning to introduce a status effect to maintain that function in another PR. Here's some work i've done on it already: https://imgur.com/a/nquyoY7

**Comment by optimumtact** [comment]
This adds too much code complexity for something that is already a toggle.

**Pull request closed without merging**